### PR TITLE
Added WorkerThread class

### LINKED ISSUE #1284: Implement a wrapper class "WorkerThread"
This should use a ThreadPool with one thread as backend.

### PR DESCRIPTION
**Pull request by marcusnaslund**
Fixes #1284

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Peer review/Agree this is a good idea @emilwestergren ?

File: test/concurrent/WorkerThreadTest.ooc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		this add("Create and wait for all", func {
+			thread := WorkerThread new()
+			thread getPromise(this taskUp) wait() . free()
+			expect(this sum, is greater than(0))
+
+			thread add(this taskUp)
+			thread add(this taskUp)
+			thread add(this taskDown)
+			thread add(this taskDown)
+			thread add(this taskUp)
```
Comment (sebastianbaginski) [inline test/concurrent/WorkerThreadTest.ooc]: Maybe verify that these tasks really run in the same thread, if this is the purpose of this class.

**Comment by emilwestergren** [comment]
Looks good

File: test/concurrent/WorkerThreadTest.ooc (hunk 1 above)
Comment (marcusnaslund) [inline test/concurrent/WorkerThreadTest.ooc]: That's a good idea, and it's now done.

**Comment by marcusnaslund** [comment]
You can perhaps take over peer review @sebastianbaginski

File: test/concurrent/WorkerThreadTest.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline test/concurrent/WorkerThreadTest.ooc]: ok, nice

**Comment by sebastianbaginski** [comment]
I agree with Emil, it looks ok.
It seems a little weird to have a thread `pool` with just one thread (as `WorkerThread` **is a** `ThreadPool` and can be used whenever ThreadPool is expected) but I understand why you want to do this.

**Comment by emilwestergren** [comment]
I agree that it should probably not *be* a `ThreadPool`, rather *have* a `ThreadPool` as backend. However, when inheriting you get the same interface for free which is nice in this case.

**Comment by marcusnaslund** [comment]
This works locally but PRInt just stops after the license checker for... some reason.

**Comment by marcusnaslund** [comment]
This fails with `gpuOff` for no appearent reason. :/ I'm taking this off final review until I sort it out.

**Pull request merged**